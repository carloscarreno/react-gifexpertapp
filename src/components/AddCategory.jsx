import React, { useState } from 'react'
export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState(''); 

  const onInputValueChange =({target})=>{ //desestructura el target desde el event
      setInputValue(target.value);  // actualiza el valor de la caja de texto
  }

  const onSubmit = (event) => {
     event.preventDefault(); //previene el refresh del navegador por el formulario
     if(inputValue.trim().length <=1 )  // la entrada debe ser mas de un caracter
         return;
     //setCategories(categories => [inputValue, ...categories]);     // actualiza la lista con  el hook
     onNewCategory(inputValue.trim());
     setInputValue('');
  }

  // Se dispara cuando presionas enter
  return (
    <form onSubmit={ onSubmit}>  
        <input
            type="Text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={onInputValueChange}
        /> 
     </form>
  )
}
