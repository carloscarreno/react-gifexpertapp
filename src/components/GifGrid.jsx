
import { useState, useEffect } from "react";
//import { getGifs } from "../helpers/getGifs";
import { GitItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({category}) => {
    // construccion de un hook personalizado
    const { images, isLoading } = useFetchGifs(category); 

   // console.log({images, isLoading});

    //const [counter, setCounter] = useState(10);
    //    const [images, setImages] = useState([]);
    
    //    const getImages = async() => {
    //       const newImages = await getGifs(category);
    //       setImages(newImages);
    //    }

    //    useEffect( () => { 
    //       getImages();
    //    }, [] );

   return (
     <>
       <h3>{category}</h3>
       {
         isLoading && ( <h2>Cargando...</h2>)
       }
       <div className="card-grid">
         { images.map( (image) => (
            <GitItem key={image.id} {...image}></GitItem>
         ))}         
       </div>
     </>   
   )
}

//  <h5>{ counter}</h5>
//   <button onClick={() => setCounter(counter + 1 )}>+1</button>
