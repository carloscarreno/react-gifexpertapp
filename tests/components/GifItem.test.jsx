
import { render , screen} from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';


describe('Pruebas de <GifItem>', () => { 
 
    const title ='Godwana';
    const url = 'https://godwana.com/godwana.gif';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
     });


    test('debe mostrar la imagen con el url y el alt', () => { 
        render(<GifItem title={title} url={url} />);
        screen.debug();
        const { src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);      

      });

    test('debe mostrar el titulo', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
      });

});

