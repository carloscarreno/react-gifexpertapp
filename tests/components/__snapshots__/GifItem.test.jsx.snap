// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifItem> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Godwana"
      src="https://godwana.com/godwana.gif"
    />
    <p>
      Godwana
    </p>
  </div>
</div>
`;
